lib/rubocop/cop/style/redundant_map_compact.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for usage of `select` or `reject`
+      # instead of `map { |value| simple_condition? ? value : next }.compact`.
+      #
+      # @example
+      #
+      #   # bad
+      #   array.map { |e| some_condition? ? e : next }.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     if some_condition?
+      #       e
+      #     else
+      #       next
+      #     end
+      #   end.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     next if some_condition?
+      #
+      #     e
+      #   end.compact
+      #
+      #   # bad
+      #   array.map do |e|
+      #     e if some_condition?
+      #   end.compact
+      #
+      #   # good
+      #   array.select { |e| some_condition? }
+      #
+      #   # good
+      #   array.reject { |e| some_condition? }
+      #
+      class RedundantMapCompact < Base
```

Comment:
I have a feeling of strangeness in the name `Redundant`.  ``Style/MapCompactWithConditionalBlock`` (or `Style/MapWithConditionalBlock`) may be close to the behavior of this cop. Of course there seems to be a better name.

Reply:
Thank you. Actually, I wasn't sure about this name.